network/handlers/drain.go

```diff
@@ -83,6 +84,19 @@ func (d *Drainer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 		return
 	}
+	if network.IsKProbe(r) {
+		if d.draining() {
+			http.Error(w, "shutting down", http.StatusServiceUnavailable)
+		} else {
+			hh := r.Header.Get(network.HashHeaderName)
+			if hh == "" {
+				http.Error(w, fmt.Sprintf("a probe request must contain a non-empty %q header", network.HashHeaderName), http.StatusBadRequest)
+				return
+			}
+			w.Header().Set(network.HashHeaderName, hh)
+			w.WriteHeader(http.StatusOK)
```

Comment:
Can we factor a function for this?  We'll want to deduplicate this logic (incl. constants) when it lands in knative/networking, so we should make this reusable.

Reply:
done😁